tests/integration/test_charm.py

```diff
@@ -127,6 +127,7 @@ async def test_consistent_data_replication_across_cluster(ops_test: OpsTest) ->
 
 @pytest.mark.order(3)
 @pytest.mark.abort_on_fail
+@pytest.mark.ha_tests
```

Comment:
might be nice to have. a different file since there will be _a lot_ of HA tests, but I am not opinionated on this right now :)

Reply:
we will address this comment in a different PR when the need arises (when we have added more HA tests)